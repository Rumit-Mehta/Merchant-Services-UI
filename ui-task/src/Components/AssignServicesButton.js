import React from "react";
import Button from "@material-ui/core/Button";
import { Alert } from "@material-ui/lab";
import Snackbar from "@material-ui/core/Snackbar";
import { newSelected } from "./ServicesTable";

const styles = {
  background: "#1176b6",
  borderRadius: 100,
  border: 0,
  color: "white",
  height: 48,
  padding: "0 30px",
};

function displayServices() {
  let servicesSelected = [];
  if (newSelected.length === 0) {
    return "No service selected";
  } else if (newSelected.length === 5) {
    return "All services selected";
  } else {
    servicesSelected = newSelected;
  }
  return "Selected : " + servicesSelected;
}

function snackbarColor() {
  if (newSelected.length === 0) {
    return "error";
  } else {
    return "success";
  }
}

export default function SimpleSnackbar() {
  const [open, setOpen] = React.useState(false);

  const handleClick = () => {
    setOpen(true);
  };

  const handleClose = (event, reason) => {
    setOpen(false);
  };

  return (
    <div>
      <Button style={styles} onClick={handleClick}>
        Assign Services
      </Button>
      <Snackbar open={open} autoHideDuration={2000} onClose={handleClose}>
        <Alert onClose={handleClose} severity={snackbarColor()}>
          {displayServices()}
        </Alert>
      </Snackbar>
    </div>
  );
}
