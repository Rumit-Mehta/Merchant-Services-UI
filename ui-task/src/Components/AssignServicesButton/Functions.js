import React from "react";
import serviceData from "../../Data/ServicesData.json";
import ServiceTableFunctions from "../ServicesTable/Functions";
import { newSelected } from "../ServicesTable/Functions";

export default () => {
  // Code to find the maximum amount of services got from ServiceData.json file
  let maxServices = serviceData.length;

  const [open, setOpen] = React.useState(false);

  const handleClick = () => {
    setOpen(true);
  };

  const handleClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }
    setOpen(false);
  };

  // Function that displays different messages on the snackbar depending on the amount of options selected
  function displayServices() {
    let servicesSelected = [];
    if (newSelected.length === 0) {
      return "No service selected";
    } else if (newSelected.length === maxServices) {
      return "All services selected";
    } else {
      servicesSelected = newSelected;
    }
    return "Selected : " + servicesSelected;
  }

  // Function that changes the snackbar colour to red (error) if no options are selected
  function snackbarColor() {
    if (newSelected.length === 0) {
      return "error";
    } else {
      return "success";
    }
  }

  return {
    open,
    setOpen,
    handleClose,
    handleClick,
    displayServices,
    snackbarColor,
  };
};
