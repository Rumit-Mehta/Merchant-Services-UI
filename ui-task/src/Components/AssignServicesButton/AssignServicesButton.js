import React from "react";
import Button from "@material-ui/core/Button";
import { Alert } from "@material-ui/lab";
import Snackbar from "@material-ui/core/Snackbar";
import Functions from "./Functions";

// button style
const styles = {
  background: "#1176b6",
  borderRadius: 100,
  border: 0,
  color: "white",
  height: 48,
  padding: "0 30px",
};

// Default function, returns button with onclick and snackbar functionality
export default function SimpleSnackbar(prop) {
  const {
    displayServices,
    handleClick,
    handleClose,
    open,
    snackbarColor,
  } = Functions();
  return (
    <div data-testid="assignServiceDiv" className="serviceDiv">
      <Button
        className="assignServiceButton"
        style={styles}
        onClick={handleClick}
      >
        Assign Services
      </Button>
      <Snackbar open={open} autoHideDuration={2000} onClose={handleClose}>
        <Alert onClose={handleClose} severity={snackbarColor()}>
          {displayServices()}
        </Alert>
      </Snackbar>
    </div>
  );
}
