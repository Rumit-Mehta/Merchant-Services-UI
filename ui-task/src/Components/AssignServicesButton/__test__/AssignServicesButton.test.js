import React from "react";
import AssignServicesButton from "../AssignServicesButton";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";

describe("Tests for Assign Service Button", () => {
  const mockClick = jest.fn();
  const mockCancel = jest.fn();

  test("matches snapshot", () => {
    const wrapper = shallow(
      <AssignServicesButton
        open
        handleClick={mockClick}
        handleClose={mockCancel}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("matches snapshot", () => {
    const wrapper = shallow(
      <AssignServicesButton handleClick={mockClick} handleClose={mockCancel} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
