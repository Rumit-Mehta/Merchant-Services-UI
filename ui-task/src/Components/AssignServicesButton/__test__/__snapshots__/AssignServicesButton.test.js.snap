// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Assign Service Button matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Tests for Assign Service Button matches snapshot 2`] = `ShallowWrapper {}`;
